reader/components/pdf-toolbar/PDFToolbar.jsx

```diff
@@ -1,15 +1,24 @@
-import React, { useState, useEffect } from 'react'
+import React, { useState, useEffect, useRef } from 'react'
 import Icon from '../icons/Icon'
-
 import './PDFToolbar.scss'
+import { useGlobalStore } from '../../store'
+import {
+  scrollToRelatedPapers,
+  unsetRelatedPapers,
+} from '../../store/ui/actions'
 
 const MIN_SCALE = 0.1
 const MAX_SCALE = 10.0
 const DEFAULT_SCALE_DELTA = 1.1
 
 const PDFToolbar = ({ viewer, eventBus }) => {
+  const [{ ui }, dispatch] = useGlobalStore()
+  const [readPosition, setReadPosition] = useState(0)
+  const [relatedPapersClicked, setRelatedPapersClicked] = useState(null)
+  const [isVisible, changeVisibility] = useState(false)
```

Comment:
I would prefer complex state object here instead the list of `useState` calls. It's just easier to read and understand what the state is about even if it's more lines of code.

Rect hooks are introduces to reduce complexity in cases where it's too much to have a class. However, there are cases where functional approach does not fit as well as object oriented does.

Reply:
I do agree. It doesn't have to be written as class. Instead of using `useState` several times I could have used one useReducer. I was just too lazy to rewrite it.

Reply:
That's not really good in general. However, it's okay to be lazy 🙂